proco/connection_statistics/serializers.py

```diff
@@ -39,3 +45,34 @@ class Meta:
             'connectivity_availability',
         )
         read_only_fields = fields
+
+
+class RealTimeConnectivitySerializer(serializers.ModelSerializer):
+    school = SeparatedReadWriteField(write_field=serializers.IntegerField(),
+                                     read_field=serializers.SerializerMethodField())
+
+    class Meta:
+        model = RealTimeConnectivity
+        fields = (
+            'school',
+            'connectivity_speed',
+            'connectivity_latency',
+            'created',
+        )
+        read_only_fields = ('created', )
+
+    def get_school(self, instance):
+        from proco.schools.serializers import BaseSchoolSerializer
+        return BaseSchoolSerializer(instance).data['name']
+
+    def validate_school(self, school_id):
+        schools = School.objects.filter(external_id=school_id)
+        if not schools.exists():
+            raise ValidationError(_('This school does not exist.'))
+        return school_id
+
+    def save(self, **kwargs):
+        school_external_id = self.validated_data.pop('school', None)
+        school = School.objects.get(external_id=school_external_id)
```

Comment:
you can just return validated school instance in validate_school to avoid duplicated db queries